File: php-pr/51/php/code-quality-naming/TaxCalculator.php
<?php

class TaxCalculator
{
    // {fact rule=code-quality-naming@v1.0 defects=0}
    public function performCalculation($firstOperand, $secondOperand, $operation)
    {
        switch ($operation) {
            case '+':
                return $firstOperand + $secondOperand;
            case '-':
                return $firstOperand - $secondOperand;
            case '*':
                return $firstOperand * $secondOperand;
            case '/':
                return $secondOperand != 0 ? $firstOperand / $secondOperand : 0;
            default:
                throw new InvalidArgumentException("Unsupported operation: {$operation}");
        }
    }
    // {/fact}
}

File: ruby-pr/53/ruby/code-quality-naming/tax_calculator.rb
class TaxCalculator
  # {fact rule=code-quality-naming@v1.0 defects=0}
  def perform_calculation(first_operand, second_operand, operation)
    case operation
    when '+'
      first_operand + second_operand
    when '-'
      first_operand - second_operand
    when '*'
      first_operand * second_operand
    when '/'
      second_operand != 0 ? first_operand / second_operand : 0
    else
      raise ArgumentError, "Unsupported operation: #{operation}"
    end
  end
  # {/fact}
end